platform/security/handler/security-handler-api/src/main/java/org/codice/ddf/security/handler/api/GuestAuthenticationToken.java

```diff
@@ -27,33 +35,52 @@
   public static final String GUEST_TOKEN_VALUE_TYPE =
       BSTAuthenticationToken.BST_NS + BSTAuthenticationToken.TOKEN_VALUE_SEPARATOR + BST_GUEST_LN;
 
-  public GuestAuthenticationToken(String realm, String ip) {
-    super(new GuestPrincipal(ip), GUEST_CREDENTIALS, realm);
+  public GuestAuthenticationToken(String realm, String name) {
+    super(new GuestPrincipal(name), GUEST_CREDENTIALS, realm);
     setTokenValueType(BSTAuthenticationToken.BST_NS, BST_GUEST_LN);
     setTokenId(BST_GUEST_LN);
 
-    if (!StringUtils.isEmpty(ip)) {
-      SecurityLogger.audit("Guest token generated for IP address: " + ip);
+    if (!StringUtils.isEmpty(name)) {
+      SecurityLogger.audit("Guest token generated for IP address: " + name);
+    }
+  }
+
+  public static String parseAddressFromName(String fullName) {
+    if (!StringUtils.isEmpty(fullName)) {
+      String[] parts = fullName.split(NAME_DELIMITER);
+      if (parts.length == 2) {
+        return formatIpAddress(parts[1]);
+      }
     }
+    return null;
+  }
+
+  // IPv6 addresses should be contained within brackets to conform
+  // to the spec IETF RFC 2732
+  private static String formatIpAddress(String ipAddress) {
+    try {
+      if (InetAddress.getByName(ipAddress) instanceof Inet6Address && !ipAddress.contains("[")) {
```

Comment:
✏️ `InetAddress.getByName()` is a costly operation in Java as it involves the DNS to resolve hosts to ip addresses.